### PR TITLE
Database

### PR DESCRIPTION
**Pull request by Shilpi75**

### CODE REVIEW

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    private Date lastTriggered;
+    @ColumnInfo(name = "snoozed_at")
+    private Long snoozedAt;
+    @ColumnInfo(name = "date_added")
+    private Date dateAdded;
+
+
+    public Task() {
+    }
+    @Ignore
```
Comment (vermayash8) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java]: Other 2 models classes have `@Ignore` on the default constructor. Are you sure that this goes on the parameterized constructor here ?

**Comment by vermayash8** [comment]
Also, please add `TasksRepository` class if possible.

File: app/build.gradle [hunk 2]
```diff
@@ -9,6 +9,7 @@ android {
         targetSdkVersion 27
         versionCode 9
         versionName "3.7"
+        multiDexEnabled true
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/build.gradle]: Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/dao/AttachmentDao.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package app.tasknearby.yashcreations.com.tasknearby.dao;
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/dao/AttachmentDao.java]: Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java (hunk 1 above)
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java]: This doesn't matter. There should be one constructor without ignore.

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java [hunk 4]
```diff
@@ -1,8 +1,19 @@
...
 
+import java.util.Date;
+
+import app.tasknearby.yashcreations.com.tasknearby.dao.AttachmentDao;
+import app.tasknearby.yashcreations.com.tasknearby.dao.LocationDao;
+import app.tasknearby.yashcreations.com.tasknearby.dao.TaskDao;
+import app.tasknearby.yashcreations.com.tasknearby.database.AppDatabase;
+import app.tasknearby.yashcreations.com.tasknearby.models.Attachment;
+import app.tasknearby.yashcreations.com.tasknearby.models.Location;
+
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java]: Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+    @ColumnInfo(name = "task_name")
+    private String taskName;
+    @ColumnInfo(name = "location_id")
+    private long locationId;
+    @ColumnInfo(name = "image_uri")
+    private String imageUri;
+    @ColumnInfo(name = "is_done")
+    private int isDone;
+    @ColumnInfo(name = "isAlarmSet")
+    private int is_alarm_set;
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java]: Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+    private Date endDate;
+    @ColumnInfo(name = "next_start_date")
+    private Date nextStartDate;
+    @ColumnInfo(name = "repeat_type")
+    private int repeatType;
+    @ColumnInfo(name = "movement_type")
+    private int movementType;
+    private int activity;
+    @ColumnInfo(name = "last_distance")
+    private float last_distance;
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java]: Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+    @ColumnInfo(name = "start_date")
+    private Date startDate;
+    @ColumnInfo(name = "end_date")
+    private Date endDate;
+    @ColumnInfo(name = "next_start_date")
+    private Date nextStartDate;
+    @ColumnInfo(name = "repeat_type")
+    private int repeatType;
+    @ColumnInfo(name = "movement_type")
+    private int movementType;
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java]: Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Location.java [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    @ColumnInfo(name = "place_name")
+    private String placeName;
+    private String latitude;
+    private String longitude;
+    @ColumnInfo(name = "use_count")
+    private int useCount;
+    @ColumnInfo(name = "is_hidden")
+    private int isHidden;
+    @ColumnInfo(name = "date_added")
+    private Date dateAdded;
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Location.java]: Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Attachment.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * Created by shilpi on 26/12/17.
+ */
+
+@Entity(tableName = DbConstants.ATTACHMENT_TABLE_NAME)
+public class Attachment {
+    @PrimaryKey(autoGenerate = true)
+    private long id;
+    private String content;
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Attachment.java]: Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/dao/LocationDao.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    @Update
+    public void updateLocation(Location location);
+
+    @Delete
+    public void deleteLocation(Location location);
+
+    @Query("SELECT * FROM " + DbConstants.LOCATION_TABLE_NAME)
+    public List<Location> getAllLocations();
+
+    @Query("SELECT * FROM " + DbConstants.LOCATION_TABLE_NAME + " WHERE id = :locationId")
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/dao/LocationDao.java]: Done

[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/dao/LocationDao.java]: Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/constants/DbConstants.java [hunk 11]
```diff
@@ -0,0 +1,19 @@
+package app.tasknearby.yashcreations.com.tasknearby.constants;
+
+/**
+ * Created by shilpi on 26/12/17.
+ */
+
+public class DbConstants {
+
+    //Database name
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/constants/DbConstants.java]: Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/constants/DbConstants.java [hunk 12]
```diff
@@ -0,0 +1,19 @@
+package app.tasknearby.yashcreations.com.tasknearby.constants;
+
+/**
+ * Created by shilpi on 26/12/17.
+ */
+
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/constants/DbConstants.java]: Done

**Pull request merged**